docs/developing.rst

```diff
@@ -780,6 +780,29 @@ needs to be transferred to the structure-of-arrays (SoA) storage in
 ``RSoA``. This is done by the ``update`` method. In the future the
 interface may change to use functions to set and retrieve positions so
 the SoA transformation of the particle data can happen automatically.
+For now, it is crucial to call either ``P.update()``, or ``P.setCoordinates(P.R)`` to populate ``RSoA``. Otherwise, the distance tables associated with ``R`` will be uninitialized or out-of-date.
+
+::
+  const SimulationCell sc;
+  ParticleSet elec(sc), ions(sc);
+  elec.setName("e");
+  ions.setName("ion0");
+
+  // initialize ions
+  ions.create(2);
+  ions.R[0] = {0.0, 0.0, 0.0};
+  ions.R[1] = {0.5, 0.5, 0.5};
+  ions.setCoordinates(ions.R); // transfer to RSoA
```

Comment:
I didn't realize that this line was needed.
Without it, the distance table contains garbage.

Reply:
Should avoid calling ``P.setCoordinates(P.R)``. Calling an public API with an internal data is bad. The documentation was clear, calling "P.update()" is necessary. Doesn't it work?

Reply:
`ions.update()` does work here.
I saw the `setCoordinates` example in `test_polynomial_eeI_jastrow.cpp` so I thought it was OK.